torch_frame/data/multi_tensor.py

```diff
@@ -143,9 +143,8 @@ def _normalize_index(
         if isinstance(index, int):
             if index < 0:
                 index = index + max_entries
-            if is_slice_end and index < 0 or index > max_entries:
-                raise IndexError(f"{idx_name} index out of bounds!")
-            elif (not is_slice_end) and (index < 0 or index >= max_entries):
+            index = min(index, max_entries)
+            if index < 0:
```

Comment:
Is it normal to clamp in this way? why are we removing `is_slice_end`?

Reply:
I see. I only checked the narrow code path, which is broken due to these checks (see the test).